src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellSectionRenderer.cs

```diff
@@ -466,11 +467,49 @@ protected virtual void UpdateTabBarItem()
 
 			ShellSection.Icon.LoadImage(ShellSection.FindMauiContext(), icon =>
 			{
-				TabBarItem = new UITabBarItem(ShellSection.Title, icon?.Value, null);
+				TabBarItem = new UITabBarItem(ShellSection.Title, GetTabBarItemIcon(icon?.Value), null);
 				TabBarItem.AccessibilityIdentifier = ShellSection.AutomationId ?? ShellSection.Title;
 			});
 		}
 
+		// https://developer.apple.com/design/human-interface-guidelines/components/navigation-and-search/tab-bars/
+		UIImage GetTabBarItemIcon(UIImage source)
+		{
+			CGSize imageSize = CGSize.Empty;
+
+			var scale = UIScreen.MainScreen.Scale;
+
+			if (scale == 3)
+				imageSize = new CGSize(51, 51);
+
+			if (scale == 2)
+				imageSize = new CGSize(34, 34);
+
```

Comment:
What happens on a scale 4 or scale 1 device? I also see these sizes are for compact, square icons - which may not be the case for Mac Catalyst and if the icon is round or something.

I also see the base size is 17, so maybe instead of if checks, we can just multiply `17 * scale` where 17 is a const with a cool, descriptive name?

Reply:
I also see 23 is a number for "regular" which may be Mac Catalyst... Not sure what the transiiton point is for the compact/regular.

Reply:
> base size is 17,

Yes, we can simplify it just multiplying.